src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -642,16 +650,18 @@ private User impersonate(final SecurityRequest request, final User originalUser)
             final boolean isDebugEnabled = log.isDebugEnabled();
             // loop over all http/rest auth domains
             for (final AuthDomain authDomain : restAuthDomains) {
-                final AuthenticationBackend authenticationBackend = authDomain.getBackend();
+                if (!(authDomain.getBackend() instanceof ImpersonationBackend impersonationBackend)) {
```

Comment:
Is this a typo here? Shorten to `if (!(authDomain.getBackend() instanceof ImpersonationBackend)) {`

Reply:
TIL of Pattern Matching introduced in JDK 17: https://docs.oracle.com/en/java/javase/17/language/pattern-matching-instanceof.html